components/search/SearchSuggestion.vue

```diff
@@ -0,0 +1,498 @@
+          nftResult.push({
+            ...nftList[i],
+            ...meta,
+            image:
+              (nftList[i]?.metadata &&
+                imageLinks[fastExtract(nftList[i].metadata)]) ||
+              getSanitizer(meta.image || '')(meta.image || ''),
+            animation_url: getSanitizer(meta.animation_url || '')(
+              meta.animation_url || ''
+            ),
+          })
+        }).then(() => {
+          this.nftResult = nftResult
+          this.isNFTResultLoading = false
+        })
+      })
+    } catch (e) {
+      logError(e, (msg) =>
+        this.$consola.warn('[PREFETCH] Unable fo fetch', msg)
+      )
+      this.isNFTResultLoading = false
+    }
+    try {
+      const query = await resolveQueryPath(
+        this.client,
+        'collectionListWithSearch'
+      )
+
+      const collectionResult = this.$apollo.query({
+        query: query.default,
+        client: this.client,
+        variables: {
+          first: this.searchSuggestionEachTypeMaxNum,
+          offset: 0,
+          denyList,
+          orderBy: this.query.sortByMultiple?.length
+            ? this.query.sortByMultiple
+            : undefined,
+          search: this.buildSearchParam(),
+        },
+      })
+
+      const {
+        data: { collectionEntities },
+      } = await collectionResult
+      const collections = unwrapSafe(
+        collectionEntities.slice(0, this.searchSuggestionEachTypeMaxNum)
+      )
+      const metadataList: string[] = collections.map(mapNFTorCollectionMetadata)
+      getCloudflareImageLinks(metadataList).then((imageLinks) => {
+        const collectionResult: CollectionWithMeta[] = []
+        processMetadata<CollectionWithMeta>(metadataList, (meta, i) => {
+          collectionResult.push({
+            ...collections[i],
+            ...meta,
+            image:
+              (collections[i]?.metadata &&
+                imageLinks[fastExtract(collections[i].metadata)]) ||
+              getSanitizer(meta.image || '')(meta.image || ''),
+          })
+        }).then(() => {
+          this.collectionResult = collectionResult
+          this.isCollectionResultLoading = false
+        })
+      })
+    } catch (e) {
+      logError(e, (msg) =>
+        this.$consola.warn('[PREFETCH] Unable fo fetch', msg)
+      )
+      this.isCollectionResultLoading = false
+    }
+  }
+
+  @Watch('activeSearchTab')
+  watchActiveSearchTab() {
+    this.resetSelectedIndex()
+  }
+
+  @Watch('name')
+  onValueChange(value) {
+    this.updateSuggestion(value)
+    this.resetSelectedIndex()
+  }
+}
+</script>
+
+<style lang="scss">
+@import '@/styles/abstracts/variables';
+
+.search-suggestion-container {
+  .selected-item {
+    box-sizing: border-box;
+    border: 1px solid $primary !important;
+  }
+
+  .link-item {
+    border: 1px solid transparent;
+    cursor: pointer;
+  }
+
+  .media-content {
+    .name {
+      max-width: 34ch;
+      overflow: hidden;
+      text-overflow: ellipsis;
+    }
+  }
```

Comment:
This file still has 200+ lines of code, would it be possible to refactor it further?

Reply:
Updated. Though there is no big stuff I could refactor in this file.